### PR TITLE
Issue 737 - Upgrade to Unity 2017.3

### PR DESCRIPTION
**Pull request by JamesMilnerUK**

### CODE REVIEW

File: frontend/globals/d.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+declare global {
```
Comment (carmenfan) [inline frontend/globals/d.ts]: Any reason this ts is called d.ts?
Just not exactly clear what this does (to my non frontend developer brain at least!)

Comment (JamesMilnerUK) [reply frontend/globals/d.ts]: This basically tells TypeScript that these are global types that it shouldn't need to complain about when it sees them in files in this folder

Comment (carmenfan) [reply frontend/globals/d.ts]: I get what the code is doing, I'm questioning the naming of this file XD

Comment (JamesMilnerUK) [reply frontend/globals/d.ts]: I did a lot of rejiging in the TypeScript Home branch, it's called overwrites.d.ts there :)

**Pull request merged**